src/link/C/zig.h

```diff
@@ -356,6 +357,18 @@ static inline long long zig_subw_longlong(long long lhs, long long rhs, long lon
     return (long long)(((unsigned long long)lhs) - ((unsigned long long)rhs));
 }
 
+static inline float zig_bitcast_uint32_t_to_float(uint32_t arg) {
+    float dest;
+    memcpy(&dest, &arg, sizeof dest);
```

Comment:
Question: isn't something like dest = *((float *)&arg) supposed to work?

Reply:
Many compilers will generate correct code because this is a common idiom, but it is technically UB.  Because of strict aliasing rules, `float *` may not alias `uint32_t *`.  The compiler has a thorough understanding of memcpy though, and will not actually call a function here.  See also https://en.cppreference.com/w/cpp/language/reinterpret_cast

Reply:
(FWIW this is well-defined in Zig because our aliasing rules are less strict.  This will make it extra difficult to generate correct C code.)

Reply:
How about doing this?

```
union { float f; uint32_t u } cast;
cast.u = arg;
return cast.f;
```

Type-punning is legal in C99 to reintepret bits and I presume we are targeting C99 at a minimum, as we rely on stdint at this time.

Reply:
See my test case below. We need a way to lower it in a way that will work inside a global variable initializer.

Reply:
The union approach is also UB in C++, for the same reason, but both are well defined in C99.  So if we are targeting C99 to be compiled only by a C compiler (not a C++ compiler), this is fine.

Re: lowering, one approach would be to generate the global for this value as a uint32_t, but then generate references to this value as stub code that converts it to a float at runtime.  (and similarly, taking the address of this value gets an untyped char pointer that will be memcpy'd into any float values when read).